.github/workflows/build_and_run_idf_examples.yml

```
@@ -0,0 +1,76 @@
+name: ESP-IDF USB Device examples
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    strategy:
```

Worth adding `fail-fast false` strategy. In case one of the builds from the matrix fails, Github acitons will not cancel the whole job.

```suggestion
    strategy:
      fail-fast: false
```